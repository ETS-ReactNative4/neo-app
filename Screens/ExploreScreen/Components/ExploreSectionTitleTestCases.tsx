import React from "react";
import { ITestCase } from "../../../TypeInterfaces/TestCases/ITestCases";
import ExploreSectionTitle from "./ExploreSectionTitle";

const ExploreSectionTitleTestCases: ITestCase[] = [
  {
    title: "Explore Section Title",
    Component: (
      <ExploreSectionTitle
        title={"HANDPICKED FOR YOU"}
        description={"Let’s tick Asia off your bucket list :)"}
      />
    )
  },
  {
    title: "Explore Section Title With Different Color",
    Component: (
      <ExploreSectionTitle
        title={"HANDPICKED FOR YOU"}
        description={"Let’s tick Asia off your bucket list :)"}
        titleColor={"rgba(255, 137, 51, 1)"}
      />
    )
  }
];

export default ExploreSectionTitleTestCases;
