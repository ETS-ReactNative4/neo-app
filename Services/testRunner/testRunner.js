import renderer from "react-test-renderer";

const testRunner = testCase => {
  test(testCase.title, () => {
    const tree = renderer.create(testCase.Component).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

export default testRunner;
