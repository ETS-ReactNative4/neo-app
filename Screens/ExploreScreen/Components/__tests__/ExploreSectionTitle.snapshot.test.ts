import ExploreSectionTitleTestCases from "../ExploreSectionTitleTestCases";
import testRunner from "../../../../Services/testRunner/testRunner";

ExploreSectionTitleTestCases.map(testRunner);
