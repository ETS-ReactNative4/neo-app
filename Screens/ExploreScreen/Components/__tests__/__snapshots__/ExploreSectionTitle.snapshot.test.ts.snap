// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Explore Section Title 1`] = `
<View>
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "fontFamily": "SourceSansPro-SemiBold",
          "fontSize": 14,
          "lineHeight": 14,
          "marginBottom": 8,
        },
        Object {
          "color": "rgba(44, 47, 66, 1)",
        },
      ]
    }
  >
    HANDPICKED FOR YOU
  </Text>
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "rgba(44, 47, 66, 1)",
          "fontFamily": "SourceSansPro-Regular",
          "fontSize": 14,
          "lineHeight": 14,
        },
        undefined,
      ]
    }
  >
    Let’s tick Asia off your bucket list :)
  </Text>
</View>
`;

exports[`Explore Section Title With Different Color 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 16,
      },
      undefined,
    ]
  }
>
  <Unknown
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "fontFamily": "SourceSansPro-SemiBold",
          "fontSize": 14,
          "lineHeight": 14,
          "marginBottom": 8,
        },
        Object {
          "color": "rgba(255, 137, 51, 1)",
        },
      ]
    }
  >
    HANDPICKED FOR YOU
  </Unknown>
  <Unknown
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "rgba(44, 47, 66, 1)",
          "fontFamily": "SourceSansPro-Regular",
          "fontSize": 14,
          "lineHeight": 14,
        },
        undefined,
      ]
    }
  >
    Let’s tick Asia off your bucket list :)
  </Unknown>
</View>
`;
